src/main/java/bisq/core/dao/node/parser/BlockParser.java

```diff
@@ -134,8 +134,10 @@ private void validateIfBlockIsConnecting(RawBlock rawBlock) throws BlockNotConne
         }
     }
 
-    private boolean isBlockNotAlreadyAdded(RawBlock rawBlock) {
-        return !bsqStateService.getBlockAtHeight(rawBlock.getHeight()).isPresent();
+    private boolean isBlockAlreadyAdded(RawBlock rawBlock) {
+        // TODO(chirhonul): shouldn't we verify that we know about the blockHash, not just that the
+        // block heights are the same? how do we handle chainsplits otherwise?
```

Comment:
Yes, true.

Reply:
That seems beyond the scope of the current PR (which is a refactor with no functional changes), which is why I added it as a `TODO` here. I'd be happy to address it in a follow-up PR.

Reply:
Ok.